Readds Explosive Decompression :D

MaltVinegar opened the pull request:
## About The Pull Request

Enables explosive_decompression for auxmos
Last thing to do is to fix the bug where firelocks don't engage

## Why It's Good For The Game

Explosive decompression is fun


TODO:
     auxmos.dll still needs to be updated
     Modularization

Fixes ripped from https://github.com/shiptest-ss13/Shiptest/pull/132

## Changelog
:cl:
add: Readds Monstermos
tweak: Modifies PreCompile.sh slightly
/:cl:

MaltVinegar commented:
Fug

TheOneAndMightyRed commented on singulostation/code/modules/research/xenobiology/crossbreeding/_structures.dm:
```diff
@@ -0,0 +1,627 @@
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/human_mob = user
+	var/mob/living/simple_animal/pet/chosen_pet = pick(/mob/living/simple_animal/pet/dog/corgi,/mob/living/simple_animal/pet/dog/pug,/mob/living/simple_animal/pet/fox,/mob/living/simple_animal/pet/cat/kitten,/mob/living/simple_animal/pet/cat/space,/mob/living/simple_animal/pet,/mob/living/simple_animal/pet/penguin)
```
```suggestion
	var/mob/living/simple_animal/pet/chosen_pet = pick(/mob/living/simple_animal/pet/dog/corgi,/mob/living/simple_animal/pet/dog/pug,/mob/living/simple_animal/pet/fox,/mob/living/simple_animal/pet/cat/kitten,/mob/living/simple_animal/pet/cat/space)
```

TheOneAndMightyRed replied:
Penguin and the default pet do not have icons

TheOneAndMightyRed commented on singulostation/code/modules/research/xenobiology/crossbreeding/_structures.dm:
```diff
@@ -0,0 +1,627 @@
+		return
+	var/mob/living/carbon/human/human_mob = user
+	var/mob/living/simple_animal/pet/chosen_pet = pick(/mob/living/simple_animal/pet/dog/corgi,/mob/living/simple_animal/pet/dog/pug,/mob/living/simple_animal/pet/fox,/mob/living/simple_animal/pet/cat/kitten,/mob/living/simple_animal/pet/cat/space,/mob/living/simple_animal/pet,/mob/living/simple_animal/pet/penguin)
+	chosen_pet = new chosen_pet(get_turf(human_mob))
```
Making a rainbow slime pylon with both having bluespace and gold in it would fail to put chosen_pet in the affected_mobs list.
A simple `affected_mobs[chosen_pet] = 0` should hopefully fix that.

TheOneAndMightyRed commented on singulostation/code/modules/research/xenobiology/crossbreeding/crystalized.dm:
```diff
@@ -0,0 +1,110 @@
+
+	var/user_turf = get_turf(user)
+
+	if(!do_after(user,15 SECONDS,FALSE,user_turf))
+		return
+
+	new crystal_type(user_turf)
+	qdel(src)
```
spamming Z would effectively make a bunch of these on the same tile.
Maybe do another
```
var/obj/structure/slime_crystal/C = locate() in range(6,get_turf(user))
if(C)
	return
```
after the do_after check to prevent duplication and stacking of pylons.

TheOneAndMightyRed commented on singulostation/code/modules/research/xenobiology/crossbreeding/_structures.dm:
```diff
@@ -0,0 +1,627 @@
+	desc = "Translucent and irregular, it can duplicate matter on a whim"
+	anchored = TRUE
+	density = FALSE
+	icon = 'icons/obj/slimecrossing.dmi'
```
Wrong icon

TheOneAndMightyRed commented on singulostation/code/modules/research/xenobiology/crossbreeding/crystalized.dm:
```diff
@@ -0,0 +1,110 @@
+	name = "crystalline extract"
+	desc = "It's crystalline,"
+	effect = "crystalline"
+	icon_state = "crystalline"
```
insert a `icon =` with the new file path

Urumasi commented on singulostation/code/modules/research/xenobiology/crossbreeding/_structures.dm:
```diff
@@ -180,7 +184,21 @@ GLOBAL_LIST_EMPTY(bluespace_slime_crystals)
 			continue
 		var/datum/gas_mixture/gas = T.return_air()
 		gas.parse_gas_string(OPENTURF_DEFAULT_ATMOS)
```
```suggestion
		gas.parse_gas_string("o2=22;n2=82") // Put this in a constant somewhere
```

Urumasi commented on singulostation/code/modules/research/xenobiology/crossbreeding/_structures.dm:
```diff
@@ -180,7 +184,21 @@ GLOBAL_LIST_EMPTY(bluespace_slime_crystals)
+			deltaTemperature *= -1
+		if(deltaTemperature)
+			gas.set_temperature(gas.return_temperature() + deltaTemperature)
+			air_update_turf()
```
```suggestion
			T.air_update_turf()
```

MaltVinegar commented:
test

MaltVinegar commented:
test

MaltVinegar commented:
test

MaltVinegar commented:
test

MaltVinegar commented:
test

MaltVinegar commented:
test

KubeRoot committed:
Switch to Putnam's auxmos repository, enable monstermos

Kube's first commit

KubeRoot committed:
Actually enabled monstermos on the DM side

KubeRoot committed:
Reenable opening thin firelocks by hand

KubeRoot committed:
Some DM-side changes from jupyterkat's PR on bee

KubeRoot committed:
Fix install_auxmos.sh script
Putnam uses a v0.2.8 format for the tags, whereas beestation doesn't have the v

KubeRoot committed:
Add precompiled windows auxmos

Git checkout on the wrong branch

MaltVinegar committed:
Update dependencies.sh

MaltVinegar committed:
Fixes some things

MaltVinegar committed:
Merge branch 'monstermos' of https://github.com/SinguloStation13/Singulo-3.0 into monstermos

MaltVinegar committed:
bumps auxmos back to 1.1.2

MaltVinegar committed:
Update PreCompile.sh

MaltVinegar committed:
Update dependencies.sh

MaltVinegar committed:
to appease linters

MaltVinegar committed:
Merge pull request #19 from SinguloStation13/MaltVinegar-patch-1

appeasing linters

KubeRoot commented on code/game/machinery/doors/firedoor.dm:
```diff
@@ -78,10 +78,19 @@
+		var/obj/mecha/M = AM
+		if(M.occupant && check_safety(M.occupant))
+			open()
+			return TRUE
```
I'm sorry malt...
Missing `singulostation edit` marker

KubeRoot commented on code/game/machinery/doors/firedoor.dm:
```diff
@@ -97,6 +106,23 @@
+		else
+			close()
+		return TRUE
+
```
Missing `singulostation edit` marker

KubeRoot commented on code/game/objects/effects/temporary_visuals/miscellaneous.dm:
```diff
@@ -388,6 +388,17 @@
+
+/obj/effect/temp_visual/dir_setting/space_wind/Initialize(mapload, set_dir, set_alpha = 255)
+	. = ..()
+	alpha = set_alpha
```
Missing `singulostation edit` marker

KubeRoot commented on code/modules/admin/admin_verbs.dm:
```diff
@@ -806,7 +806,6 @@ GLOBAL_PROTECT(admin_verbs_hideable)
 			continue
 		GM.parse_gas_string(F.initial_gas_mix)
 		F.copy_air(GM)
-		F.update_visuals()
```
Missing `singulostation edit` marker
Also, do you happen to know why this is removed?

MaltVinegar replied:
Not sure, it was removed in your PR
https://github.com/SinguloStation13/SinguloStation-Blue/pull/2/files#diff-46efb60da866bae636d30b19603fa3a90545244fd0f1f5b154edaf1e11b41daf

KubeRoot commented on code/modules/atmospherics/environmental/LINDA_turf_tile.dm:
```diff
@@ -131,8 +131,10 @@
+	return
 
-/turf/open/proc/update_visuals()
+/turf/open/update_visuals()
```
Missing `singulostation edit` marker

KubeRoot commented on code/modules/atmospherics/environmental/LINDA_turf_tile.dm:
```diff
@@ -191,7 +193,11 @@
+/turf/proc/consider_firelocks()
+	return
+
+/turf/open/consider_firelocks(turf/T2)
```
Missing `singulostation edit` marker

KubeRoot commented on code/modules/atmospherics/environmental/LINDA_turf_tile.dm:
```diff
@@ -243,6 +249,8 @@
 		if (!M.anchored && !M.pulledby && M.last_high_pressure_movement_air_cycle < SSair.times_fired)
 			M.experience_pressure_difference(pressure_difference * multiplier, pressure_direction, 0, pressure_specific_target)
+	if(pressure_difference > 100)
+		new /obj/effect/temp_visual/dir_setting/space_wind(src, pressure_direction, CLAMP(round(sqrt(pressure_difference) * 2), 10, 255))
```
Missing `singulostation edit` marker

KubeRoot commented on code/controllers/subsystem/air.dm:
```diff
@@ -50,11 +50,11 @@ SUBSYSTEM_DEF(air)
 
 	// Max number of turfs equalization will grab.
-	var/equalize_turf_limit = 10
+	var/equalize_turf_limit = 20
```
Missing `singulostation edit` marker

KubeRoot commented on code/controllers/subsystem/air.dm:
```diff
@@ -50,11 +50,11 @@ SUBSYSTEM_DEF(air)
 	var/equalize_hard_turf_limit = 2000
 	// Whether equalization should be enabled at all.
-	var/equalize_enabled = FALSE
+	var/equalize_enabled = TRUE
```
Missing `singulostation edit` marker

MaltVinegar commented:
When the 'Missing `singulostation edit` marker' hits

MaltVinegar committed:
Modularization

MaltVinegar committed:
stat to machine_state

MaltVinegar committed:
Revert "stat to machine_state"

This reverts commit 7d2d9380d612c7692f242d317b93172911719433.

MaltVinegar committed:
Merge branch 'master' into monstermos

MaltVinegar committed:
Merge branch 'master' into monstermos

MaltVinegar committed:
Merge branch 'master' into monstermos

MaltVinegar committed:
machine_stat moment

MaltVinegar committed:
Merge branch 'master' into monstermos

MaltVinegar committed:
Fixes config merge conflict
I though I discarded this change??

MaltVinegar committed:
Calling parent in update_icon_state

MaltVinegar committed:
emag fix

MaltVinegar committed:
Forgot to call parent

MaltVinegar committed:
allows greps to ignore directional firedoors

MaltVinegar committed:
fixes lattices inside of the asteroid

MaltVinegar committed:
removes redundent functions

Co-authored-by: KubeRoot <kubinator4321@gmail.com>

MaltVinegar committed:
Wait, no, *that* wasn't redundant

MaltVinegar committed:
Forgot this

MaltVinegar committed:
Merge branch 'monstermos' of https://github.com/SinguloStation13/SinguloStation13 into monstermos

MaltVinegar committed:
Merge commit 'refs/pull/360/head' of https://github.com/SinguloStation13/SinguloStation13 into monstermos

MaltVinegar committed:
removes adjacent air subsystem

MaltVinegar committed:
Removes ImmediateCalculateAdjacentTurfs

MaltVinegar committed:
guh

MaltVinegar committed:
some more changes

MaltVinegar committed:
:pleading:

MaltVinegar committed:
fixes ?

MaltVinegar committed:
forgot this

MaltVinegar committed:
forgot this too

MaltVinegar committed:
test fix?

MaltVinegar committed:
makes emergency_pressure_close conform to SHOULD_NOT_SLEEP

MaltVinegar committed:
Downgrades auxmos, upgrades byond, possible fix for firedoor shenanigans

MaltVinegar committed:
downgrades auxmos again

MaltVinegar committed:
??

MaltVinegar committed:
guh

MaltVinegar committed:
Revert "guh"

This reverts commit 70fa47113e6f3cc129061c33c05149e073b81144.

MaltVinegar committed:
Revert "??"

This reverts commit 8b004bbcebf109cd1a455fcc481a1052c9e99df2.

MaltVinegar committed:
Revert "downgrades auxmos again"

This reverts commit 1d2bd1a9db5172c34aa017d8e1a54d351ec00975.

MaltVinegar committed:
Revert "Downgrades auxmos, upgrades byond, possible fix for firedoor shenanigans"

This reverts commit 1d9731a8cde03c591df59d94713cd5e5b21a8149.

MaltVinegar committed:
possible fix (again)?

MaltVinegar committed:
Merge branch 'master' into monstermos

MaltVinegar committed:
Forgot to change this

MaltVinegar committed:
updates auxmos.dll

MaltVinegar committed:
modularization

MaltVinegar committed:
Merge branch 'monstermos' of https://github.com/SinguloStation13/SinguloStation13 into monstermos

MaltVinegar committed:
Update firedoor.dm

MaltVinegar committed:
Merge branch 'master' into monstermos

MaltVinegar committed:
fixes merge conflicts

MaltVinegar committed:
Merge branch 'monstermos' of https://github.com/MaltVinegar/SinguloStation13 into monstermos

MaltVinegar committed:
updates auxmos

MaltVinegar committed:
Fixes .proc references

MaltVinegar committed:
updates PreCompile script
